frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -1,223 +1,243 @@
-        isEditable={!isArchive}
-        isEditing={isEditing}
-        isSavingEdits={isSavingEdits}
-        contentWhenEditing={
+    <DashboardDescriptionListGroup
+      editButtonTestId="editable-labels-group-edit"
+      saveButtonTestId="editable-labels-group-save"
+      title={title}
+      isEmpty={labels.length === 0}
+      contentWhenEmpty={contentWhenEmpty}
+      isEditable={!isArchive}
+      isEditing={isEditing}
+      isSavingEdits={isSavingEdits}
+      contentWhenEditing={
+        <>
           <LabelGroup
-            data-testid="label-group"
+            data-testid="editable-label-group"
             isEditable={!isSavingEdits}
-            numLabels={unsavedLabels.length}
-            addLabelControl={
-              !isSavingEdits && (
-                <Label
-                  textMaxWidth="40ch"
-                  color="blue"
-                  variant="outline"
-                  isOverflowLabel
-                  onClick={toggleAddLabelModal}
-                >
-                  Add label
-                </Label>
-              )
-            }
+            numLabels={10}
+            expandedText="Show Less"
+            collapsedText="Show More"
           >
             {unsavedLabels.map((label, index) => (
               <Label
-                key={label}
-                color="blue"
-                data-testid="label"
+                data-testid={`editable-label-${label}`}
+                key={label + index}
+                color={shouldBeRed(label, index) ? 'red' : 'blue'}
                 isEditable={!isSavingEdits}
-                editableProps={{ 'aria-label': `Editable label with text ${label}` }}
                 onClose={() => removeUnsavedLabel(label)}
-                closeBtnProps={{ isDisabled: isSavingEdits }}
-                onEditComplete={(_event, newText) => {
-                  if (!reservedKeys.includes(newText) && newText.length <= 63) {
-                    editUnsavedLabel(newText, index);
-                  }
+                closeBtnProps={{
+                  isDisabled: isSavingEdits,
+                  'data-testid': `remove-label-${label}`,
+                }}
+                onEditComplete={(event, newText) => handleEditComplete(event, newText, label)}
+                editableProps={{
+                  defaultValue: label,
+                  'aria-label': 'Edit label',
+                  'data-testid': `edit-label-input-${label}`,
                 }}
               >
                 {label}
               </Label>
             ))}
+            <Button
+              data-testid="add-label-button"
+              variant="plain"
+              className="pf-v5-c-label pf-m-outline"
+              onClick={addNewLabel}
+              isDisabled={isSavingEdits}
+              style={{
+                border: '2px solid #d2d2d2',
+                color: '#0066CC',
+                backgroundColor: 'transparent',
+              }}
```

Comment:
These inline styles override the border/color at all times so there is no longer a hover state. We should avoid inline styles where possible.

I don't think we need our own Button here. Can we not use the `addLabelControl` prop of LabelGroup like we had before? That automatically adds a button with the correct hover styles while editing like in this PF example. https://www.patternfly.org/components/label#editable-label-group
I locally reverted that part of this change and put back that `addLabelControl` but just had it call your new `addNewLabel` function, and it appears to work fine.